server.js

```diff
@@ -5,10 +5,29 @@ const fs = require('fs');
 const Q = require('q');
 
 const OAuthGithub = require('./server/OAuthGithub');
-const argv = require('minimist')(process.argv.slice(2));
 const config = require('./config');
 const db = require('./database/connect.js');
 const dbHelper = new (require('./database/helper.js'))(db);
+const argv = require('commander');
+argv.description('Usage: ' + process.argv[0] + ' ' + process.argv[1] + ' [options]')
+    .option('--non-github',                    'Do not connect to the GitHub application, **OR**')
```

Comment:
Since having the `--non-github` option is complicating things, why don't we get rid of it? That should simplify things.

There could even be additional console output after the server is starting that says:
>The server will attempt to connect to Github as needed.

or

>Not connected to Github. See --help for more information.

I appreciate what you've already done, so if you don't want to spend extra time making more changes, I completely understand.